erpnext_shopify/sync_orders.py

```diff
@@ -0,0 +1,159 @@
+		and not so.per_billed:
+		si = make_sales_invoice(so.name)
+		si.shopify_id = order.get("id")
+		si.naming_series = shopify_settings.sales_invoice_series or "SI-Shopify-"
+		si.is_pos = 1
+		si.cash_bank_account = shopify_settings.cash_bank_account
+		si.submit()
+
+def create_delivery_note(order, shopify_settings, so):
+	for fulfillment in order.get("fulfillments"):
+		if not frappe.db.get_value("Delivery Note", {"shopify_id": fulfillment.get("id")}, "name") and so.docstatus==1:
+			dn = make_delivery_note(so.name)
+			dn.shopify_id = fulfillment.get("id")
+			dn.naming_series = shopify_settings.delivery_note_series or "DN-Shopify-"
+			dn.items = update_items_qty(dn.items, fulfillment.get("line_items"), shopify_settings)
+			dn.save()
+
+def update_items_qty(dn_items, fulfillment_items, shopify_settings):
+	return [dn_item.update({"qty": item.get("quantity")}) for item in fulfillment_items for dn_item in dn_items\
+		 if get_item_code(item) == dn_item.item_code]
+
+def get_discounted_amount(order):
+	discounted_amount = 0.0
+	for discount in order.get("discount_codes"):
+		discounted_amount += flt(discount.get("amount"))
+	return discounted_amount
+
+def get_item_line(order_items, shopify_settings):
+	items = []
+	for item in order_items:
+		item_code = get_item_code(item)
+		items.append({
+			"item_code": item_code,
+			"item_name": item.get("name"),
+			"rate": item.get("price"),
+			"qty": item.get("quantity"),
+			"stock_uom": item.get("sku"),
+			"warehouse": shopify_settings.warehouse
+		})
+	return items
+
+def get_item_code(item):
+	item_code = frappe.db.get_value("Item", {"shopify_id": item.get("variant_id")}, "item_code")
+	if not item_code:
+		item_code = frappe.db.get_value("Item", {"shopify_id": item.get("product_id")}, "item_code")
+
+	return item_code
+
+def get_tax_line(order, shipping_lines, shopify_settings):
+	taxes = []
+	for tax in order.get("tax_lines"):
+		taxes.append({
+			"charge_type": _("On Net Total"),
+			"account_head": get_tax_account_head(tax),
+			"description": tax.get("title") + "-" + cstr(tax.get("rate") * 100.00),
+			"rate": tax.get("rate") * 100.00,
+			"included_in_print_rate": set_included_in_print_rate(order)
+		})
+
+	taxes = update_taxes_with_shipping_rule(taxes, shipping_lines)
+
+	return taxes
+
+def set_included_in_print_rate(order):
+	if order.get("total_tax"):
+		if (flt(order.get("total_price")) - flt(order.get("total_line_items_price"))) == 0.0:
+			return 1
+	return 0
+
+def update_taxes_with_shipping_rule(taxes, shipping_lines):
+	for shipping_charge in shipping_lines:
+		taxes.append({
+			"charge_type": _("Actual"),
+			"account_head": get_tax_account_head(shipping_charge),
+			"description": shipping_charge["title"],
+			"tax_amount": shipping_charge["price"]
+		})
+
+	return taxes
+
+def get_tax_account_head(tax):
+	tax_account =  frappe.db.get_value("Shopify Tax Account", \
+		{"parent": "Shopify Settings", "shopify_tax": tax.get("title")}, "tax_account")
```

Comment:
Is it possible to search using some kind of tax id instead of tax title, as title could be changed?